LunaDll/SdlMusic/MusicManager.cpp

```diff
@@ -537,21 +558,46 @@ void MusicManager::loadCustomSounds(std::string episodePath, std::string levelCu
 void MusicManager::resetSoundsToDefault()
 {
     initArrays();
-    loadSounds(defaultSndINI, PGE_SDL_Manager::appPath);
+    initArraysSound();
+    loadSounds(defaultSndINI, PGE_SDL_Manager::appPath, false);
     loadMusics(defaultMusINI, PGE_SDL_Manager::appPath);
     rebuildSoundCache();
 }
 
+void MusicManager::resizeSoundArrays(int new_max_sound_id) {
+    bool any_change = false;
+    if (new_max_sound_id != max_soundeffect_count)
+    {
+        //Free old level music array and create a new one
+        if (sounds != NULL)
+        {
+            delete[] sounds;
+        }
+        max_soundeffect_count = new_max_sound_id;
+        sounds = new ChunkEntry[new_max_sound_id];
+    }
+    if (any_change)
+    {
+        //Force re-populate music if the size of either array changed
+        initArraysSound();
+    }
+}
 
-void MusicManager::initArrays()
+void MusicManager::initArraysSound()
 {
     curRoot = PGE_SDL_Manager::appPath;
-    for(int i=0; i<91; i++)
+    for(int i = 0; i < max_soundeffect_count; i++)
```

Comment:
This is not safe/correct code. `i` is used to index into `defaultChunksList` and `chunksChannelsList` which have a size of `defaultSoundCount`, which may be smaller than `max_soundeffect_count` when this is called.

Reply:
I just did a commit fixing the issue where channelChunksList doesn't have any values above 91. Anything above 91 will take the channel -1.

Check it and see if it's good enough, if you have the time to.